app/Providers/ModelServiceProvider.php

```diff
@@ -17,14 +17,17 @@ class ModelServiceProvider extends ServiceProvider
      */
     public function boot()
     {
-        // When Signups are saved create an event for them.
+        // When a Signup's why_participated, quantity, quantity_pending, or deleted_at are changed, create an event.
+        // @TODO: when we move quantity on the post, we'll want to remove this check below.
         Signup::saved(function ($signup) {
-            $signup->events()->create([
-                'content' => $signup->toJson(),
-                // Use the authenticated user if coming from the web,
-                // otherwise use the id of the user in the request.
-                'user' => auth()->user() ? auth()->user()->northstar_id : $signup->northstar_id,
-            ]);
+            if ($signup->isDirty('why_participated') || $signup->isDirty('quantity') || $signup->isDirty('quantity_pending') || $signup->isDirty('deleted_at')) {
```

Comment:
does this work? i thought that `isDirty()` only checks if something has been changed before it is `Model::saved()` maybe change that to `Signup::saving()`

Reply:
it did work but `Signup::saving()` works too! I changed it to that since it is more readable.

Reply:
hmm actually it seems like `Signup::saving()` breaks the tests bc 
```
21) Tests\Http\Three\SignupTest::testDeletingASignup
Illuminate\Database\QueryException: SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'eventable_id' cannot be null (SQL: insert into `events` (`content`, `user`, `eventable_id`, `eventable_type`, `updated_at`, `created_at`) values ("{\"northstar_id\":\"5570b6cea59dbf3b7a8b4567\",\"campaign_id\":1173,\"campaign_run_id\":4611,\"quantity_pending\":7921,\"why_participated\":\"Ut ut hic omnis et ut sint cum perspiciatis.\",\"source\":\"phoenix-web\",\"details\":\"fun-affiliate-stuff\"}", 5570b6cea59dbf3b7a8b4567, , Rogue\Models\Signup, 2017-10-27 14:45:20, 2017-10-27 14:45:20))
```

Reply:
I double checked with testing manually and tests are passing so am going to change back to `Signup::saved()`